ui/lib/sync/addon/utils/api-method-resolver.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { capitalize } from '@ember/string';
+
+import type { DestinationType } from 'vault/sync';
+
+type TypeKey = 'AwsSm' | 'AzureKv' | 'GcpSm' | 'Gh' | 'VercelProject';
+
+// unfortunately type is not a param for sync destination api methods
+// it seems that for documentation purposes it was intentionally set up this way
+export default function apiMethodResolver(action: 'read' | 'write' | 'delete', type: DestinationType) {
+  const formattedType = type.split('-').reduce((str, word) => str.concat(capitalize(word)), '') as TypeKey;
```

Comment:
is it necessary to make this dynamic? It seems like it'd be a little easier to read if this was just a map. Or ember's `classify` string method could be used here https://api.emberjs.com/ember/4.2/classes/String/methods/classify?anchor=classify

Reply:
Nice I don't think I've ever noticed the `classify` method! Looks like it does the same thing so I will update to use that.